doc/docs/master/how-tos/developer-workflow/build-pipelines.md

```diff
@@ -0,0 +1,118 @@
+
+1. All files (code, etc.) are copied from the build path to a PFS repository, `<pipeline name>_build`, which we can think of as the source code repository. In the case above, everything in `./source` would be copied to to the PFS `<pipeline name>_build` repository.
+
+1. A pipeline called `<pipeline name>_build` starts, reading reading files from the `<pipeline name>_build` repo and runs the `build.sh` script to pull dependencies and compile any requirements from the source code.
+
+1. The running pipeline, `<pipeline name>`, is updated (or created if it doesn't already exist) with the the new source files and built assets.
+
+!!! note
+      You can optionally specify a `.pachignore` file, which uses [ohmyglob](https://github.com/pachyderm/ohmyglob) entries to prevent certain files from getting pushed to this repo.
+
+The updated pipeline contains the following PFS repos mapped in as inputs:
+
+1. `/pfs/source` - source code that is required for running the pipeline. There are language-specific requirements on the structure of this source code mentioned in the [Python Builder](#python-builder) section.
+
+1. `/pfs/build` - any artifacts resulting from the build process.
+
+1. `/pfs/<input(s)>` - any inputs specified in the pipeline specification.
+
+## Builders
+The builder is responsible for determining what Docker image is used for the base image and what steps to run during the pipeline build and deployment. If a builder has a standard implementation (currently `python` and `go`), then only the `transform.build.language` needs to be set, as it has impelmentations for `build.sh` and `run.sh`. The order of preference is:
+
+1. `transform.build.language`
+1. `transform.build.image`
+1. `transform.image`
+
+If `transform.buil.language` is not specified, then the build pipeline is non-standard and uses the `transform.build.image` container (and subsequently the `transform.build` if `transform.build.image` is not provided). In a non-standard implementation, the convention is to provide `build.sh` and `run.sh` scripts to fulfill the build pipeline requirements. 
+
+Similarly, a build pipeline by default will run `sh /pfs/build/run.sh` by default. If a `transform.cmd` is specified, it will take precedence over `run.sh`.
+
+### Python Builder
+
+The Python builder requires a file structure similar to the following:
+
+```tree
+./map
+├── source
+│   ├── requirements.txt
+│   ├── ...
+│   └── main.py
+└── build-pipeline.json
+```
+There must exist a `main.py` which acts as the entrypoint for the pipeline. Optionally, a `requirements.txt` can be used to specify pip packages that will be installed during the build process. Other supporting files in the directory will also be copied and available in the pipeline if they are not excluded by the `.pachignore`.
+
+The `build.sh` and `run.sh` files are optional, as the Python Builder already contains these scripts. However, if the base image is modified, they must be provided.
```

Comment:
This is not true, user-specified `build.sh` and `run.sh` do not work

Reply:
It's not true if they're using the `language=python`, but isn't it true if you modify the base image to be something other than the standard Python Builder?

Reply:
(i.e. the new base image wouldn't have a `build.sh` or `run.sh`)

Reply:
Yes that's correct, but pipeline builder images are separate from the user code, and would be stored in separate locations